### PR TITLE
Rework

### PR DESCRIPTION
**Pull request by machour**
* Changed namespace
* Removed deprecated classes
* Removed non amqp drivers (see #13)
* Fixed all tests but the "retry" test

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -75,9 +72,26 @@
...
+            "type": "vcs",
+            "url": "https://github.com/yiisoft/mutex-db-mysql",
+            "no-api": true
+        },
+        {
+            "type": "vcs",
+            "url": "https://github.com/yiisoft/mutex-db-pgsql",
+            "no-api": true
+        }
+    ]
```
Comment (machour) [inline composer.json]: @samdark Could you publish those 3 packages to packagist?

File: src/Cli/Command.php [hunk 2]
```diff
@@ -211,9 +211,9 @@ protected function handleMessage($id, $message, $ttr, $attempt)
 
             return $result === self::EXEC_DONE;
         } catch (ProcessRuntimeException $error) {
-            list($job) = $this->queue->unserializeMessage($message);
+            [$job] = $this->queue->unserializeMessage($message);
 
-            return $this->queue->handleError(ExecEvent::before($id, $job, $ttr, $attempt, $error));
+            return $this->queue->handleError(ExecEvent::error($id, $job, $ttr, $attempt, $error));
```
Comment (machour) [inline src/Cli/Command.php]: This is wrong, error() takes an Event as first and only parameter.

@zhuravljov something went horribly wrong with ErrorEvent removal, and that leads to "retry" tests failing.
I'm having a little trouble debugging this as this is all happening in child processes.. could you have a look please? 🙏

File: tests/docker-compose.yml [hunk 3]
```diff
@@ -60,17 +60,6 @@ services:
     depends_on: *php_depends_on
```
Comment (machour) [inline tests/docker-compose.yml]: Do we still need docker setups? Travis seems to be handling what's left of the drivers everything pretty well.

**Comment by machour** [comment]
@samdark could you do a quick review and merge this if it looks good enough?

Keeping this PR up to date is going to be a nightmare with all the namespace changes currently taking place.

File: UPGRADE.md [hunk 4]
```diff
@@ -7,28 +7,28 @@ application when you upgrade the package from one version to another.
 Upgrade to 2.1.1
```
Comment (samdark) [inline UPGRADE.md]: These upgrade notes doesn't seem to be actual anymore.

File: composer.json (hunk 1 above)
Comment (samdark) [reply composer.json]: Done.

File: src/Drivers/Db/Queue.php [hunk 5]
```diff
@@ -249,7 +254,7 @@ private function moveExpired()
             $this->db->createCommand()->update(
                 $this->tableName,
                 ['reserved_at' => null],
-                '[[reserved_at]] < :time - [[ttr]] and [[reserved_at]] is not null and [[done_at]] is null',
+                '[[reserved_at]] < :time - [[ttr]] and [[done_at]] is null',
```
Comment (samdark) [inline src/Drivers/Db/Queue.php]: Is there a reason for this change?

File: tests/docker-compose.yml (hunk 3 above)
Comment (samdark) [reply tests/docker-compose.yml]: Maybe for local tests...

File: src/Drivers/Db/Queue.php (hunk 5 above)
Comment (machour) [reply src/Drivers/Db/Queue.php]: This was a backport of https://github.com/yiisoft/yii2-queue/commit/eb8c7d4c635ae9da1c09e4fb3ae41452e33bed8b

**Comment by samdark** [comment]
Looks good enough to me. @machour

**Pull request merged**